server/src/utils/fight/fightMethods.ts

```diff
@@ -981,31 +1019,39 @@ const activateSuper = (
         // Change pet owner
         pet.master = fighter.id;
         pet.team = fighter.team;
+        // Set pet initiative to fighter initiative (to act right after)
+        pet.initiative = fighter.initiative - 0.01;
       }
 
-      // Abort if no pet hypnotised
-      if (hypnotisedPets.length === 0) return false;
+      // Hypnotize opponent's brutes and bosses
+      const opponents = getOpponents({ fightData, fighter, bruteAndBossOnly: true})
+      opponents.forEach((opponent) => {
+        opponent.hypnotized = true;
+      });
 
       // Add hypnotise step
       fightData.steps.push({
         a: StepType.Hypnotise,
         b: fighter.index,
+        t: opponents.map((opponent) => opponent.index),
         p: hypnotisedPets,
       });
 
       break;
     }
     case SkillName.flashFlood: {
       // Choose opponent
-      const opponent = getRandomOpponent({ fightData, fighter, bruteOnly: true });
+      const opponent = getRandomOpponent({ fightData, fighter, bruteAndBossOnly: true });
 
       if (!opponent) {
         return false;
       }
 
-      // Drop shield
-      if (fighter.shield) {
-        dropShield(fightData, fighter);
+      let throwShield = false;
+      // Throw shield if there are not too much weapons left
+      if (fighter.shield && fighter.weapons.length < 6) {
+        throwShield = true;
+        dropShield({ fightData, fighter, addStep: false });
       }
```

Comment:
This is not a clear enough condition.
The shield should be thrown depending on the number of uses left instead.

Reply:
**"The shield should be thrown depending on the number of uses left instead."**

This is what it does.
Shield will only be thrown if its the last use.
If we take only "super.uses === 1" as a condition, this will basically never happen as the "real" number of use depends on weapon number left and almost never on number of uses left.
I will add "super.uses === 1" as and aditional situation and clarify the comment.


**"I'd rather not reposition at all."**

This is not as simple as we think at first. It's important to note that RepositionFighter reposition _every_ fighter. The goal is not only to reposition a fighter when it plays, but also before other plays.

Not repositionning fighters can lead to 
- **prolongated** (because moveback is only called after melee attack, so half of the time the fighter will stay mispositionned after it's turn. Plus it can have a very low speed and other fighters will play a lot of turn while it stays mispositionned)
- **unreadable** (the fight is really hard to understand as there is no more side and everything is going melee, it is very peturbing especially with multiple fighters)
- **gimmical** (the fighters are moving on y axis a lot, making it look like a drunk melee fight. During others turns, they will also look really strange as they are almost touching idle ennemies and doing nothing more than "staring menacingly")
- **unrealistic** (both fighters start to throw weapons and bombs at melee range)

situations.

This will lead to people postings clips in the "bugs" section saying that the fight is buggued / unreadable.

I have tested my solution on a loot of fights and I can guarantee it looks really great and natural, and the fights are more dynamic and readable.


**"It also doesn't make sense for a fighter to moveBack before acting."**

They should move back before throwing / casting.
But you are right, they should not move back before moveTo.
I will change that.

Reply:
As I said, I'm seeing in the discord some people happy about the dropShield as they want to survive then maximize the damage of their only cast, and some other players unhappy as they want to survive until further casts...

Throwing at last use would make everyone happy...

I'm solving [this](https://discord.com/channels/693082011484684348/1306936035539943434/1306936035539943434)